### DISCUSSION TITLE
Handling the Password Reset for a user by Supabase itself.

### DISCUSSION THREAD

**Discussion by MBSA-INFINITY** [supabase/supabase#15717] [Feature Requests]
**The Problem**
The function `supabase.auth.reset_password_email(email)` only sends a reset mail to the particular email address mentioned with a `PASSWORD_RECOVERY` event but not actually handles the password reset of that account itself, unlike **firebase**. This function `reset_password_email()` should not only just send a password reset mail but also handle password reset with a unique form link specific to that user, so that the particular user can change the password. I hope that clarifies the problem.

**The Solution**
The function `reset_password_email(email)` should generate a unique `supabase form link` which would contain a form to reset the password and this link should be sent to the particular mail as a password reset mail. Once the password is reset successfully, user would be redirected back to the main application. This way supabase users wouldn't have to worry about password reset themselves by handling the `PASSWORD_RECOVERY` event. They just have to call this function and rest, supabse would handle. ***I would be more than happy to take up on this issue and contribute to the supabase community!***

***Firebase Authentication***  have this function `send_password_reset_email()` that does the same thing described above, hence as an alternative to Firebase, `Supabase` should also provide this provision!

**Additional context**
Do checkout this Loom:- [Firebase Password Reset Flow](https://www.loom.com/share/211759ff187e4a8f9dd5d14dd434c7e5)

**Comment by Rooban19** [comment]
This is an important required feature for supabse

**Comment by Akilesh1989** [comment]
This will be a very useful add on to Supabase. Looking forward to this update.

**Comment by Om25091210** [comment]
I crossed check the things and seems like we really need this kind of feature.

**Comment by iwint** [comment]
I agree, this sounds great and looks like a much easier way to reset the password

**Comment by dAnkc0der** [comment]
This seems to be a more convenient way for resetting the password. Looking forward to this update.

**Comment by schox** [comment]
Yes, it would be great to redirect to a supabase form by default and with an over-ride which does the existing functionality for those that need it.
